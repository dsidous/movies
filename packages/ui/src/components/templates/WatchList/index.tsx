/* eslint-disable no-underscore-dangle */
import _ from 'lodash';
import React from 'react';
import Alert from '@material-ui/lab/Alert';

import Section from '../../atoms/Section';
import FirebaseAuthContext from '../../contexts/FirebaseAuthContext';
import MovieCard from '../../molecules/MovieCard';
import PageTransition from '../../atoms/PageTransition';

import useStyles from './styles';

const WatchList: React.FC = () => {
  const classes = useStyles();

  return (
    <PageTransition>
      <FirebaseAuthContext.Consumer>
        {({ user, authUser }) => (
          <Section noborder disableGutters>
            <Section.Header>My Watchlist</Section.Header>
            {_.size(authUser && user?.watchlist) > 0 ? (
              <ul className={classes.list}>
                {_.map(user?.watchlist, (movie) => {
                  const media = movie.name !== '' ? 'tv' : 'movie';
                  return (
                    <MovieCard key={movie.id} movie={movie} media={media} />
                  );
                })}
              </ul>
            ) : (
              authUser && (
                <Alert severity='info'>
                  No movie or tv show added to your watchlist yet.
                </Alert>
              )
            )}
            {!authUser && (
              <Alert severity='error'>
                Please log in or register for manage your watchlist
              </Alert>
            )}
          </Section>
        )}
      </FirebaseAuthContext.Consumer>
    </PageTransition>
  );
};

export default WatchList;
