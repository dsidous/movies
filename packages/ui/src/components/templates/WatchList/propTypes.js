import { bool, object } from 'prop-types';

export const contextTypes = {
  router: object.isRequired,
};

export const propTypes = {
  configLoading: bool,
  config: object,
};

export const defaultProps = {};
