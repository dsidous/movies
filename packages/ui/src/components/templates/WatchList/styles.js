import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    list: {
      margin: '50px auto',
      padding: 0,

      [theme.breakpoints.up('sm')]: {
        display: 'grid',
        gridGap: 40,
        gridArea: 'result',
        gridTemplateColumns: 'repeat(2,minmax(200px, 1fr))',
      },
    },
  }),
  { name: 'watchlist' },
);
