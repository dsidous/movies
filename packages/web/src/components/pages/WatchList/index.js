import { WatchList } from '@movies/ui';
import { withConfig } from '@movies/common';

export default withConfig()(WatchList);
