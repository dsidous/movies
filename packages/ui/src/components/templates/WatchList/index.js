/* eslint-disable no-underscore-dangle */
/* eslint-disable camelcase */
import _ from 'lodash';
import React from 'react';

import { propTypes } from './propTypes';
import FirebaseAuthContext from '../../context/FirebaseAuthContext';
import MovieCard from '../../molecules/MovieCard';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition';

import useStyles from './styles';

const WatchList = props => {
  const { configLoading } = props;
  const classes = useStyles();

  if (configLoading) {
    return <Spinner />;
  }

  const { config } = props;
  const img_base_path =
    config.images.secure_base_url + config.images.poster_sizes[2];

  return (
    <PageTransition>
      <FirebaseAuthContext.Consumer>
        {({ user, authUser }) => (
          <div className={classes.root}>
            {_.size(authUser && user?.watchlist) > 0 ? (
              <ul className={classes.list}>
                {_.map(user.watchlist, movie => {
                  const media =
                    movie.__typename === 'Tv_detailed' ||
                    movie.__typename === 'Tv'
                      ? 'tv'
                      : 'movie';
                  return (
                    <MovieCard
                      key={movie.id}
                      movie={movie}
                      img_base_path={img_base_path}
                      media={media}
                    />
                  );
                })}
              </ul>
            ) : (
              authUser && (
                <div>No movie or tv show added to your watchlist yet.</div>
              )
            )}
            {!authUser && (
              <div>Please log in or register for manage your watchlist</div>
            )}
          </div>
        )}
      </FirebaseAuthContext.Consumer>
    </PageTransition>
  );
};

WatchList.propTypes = propTypes;

export default WatchList;
