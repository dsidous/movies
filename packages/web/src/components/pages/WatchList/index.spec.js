import React from 'react';
import { shallow } from 'enzyme';
import WatchList from '.';

const mockProps = {
  config: {
    images: {
      base_url: 'path/to/image',
      secure_base_url: 'path/to/image',
      backdrop_sizes: ['1', '2', '3', '4'],
      poster_sizes: ['1', '2', '3', '4'],
      profile_sizes: ['1', '2', '3', '4'],
      still_sizes: ['1', '2', '3', '4'],
    },
  },
};

describe('Pages/WatchList', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<WatchList {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
