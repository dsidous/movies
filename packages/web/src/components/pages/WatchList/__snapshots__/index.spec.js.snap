// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages/WatchList should render as expected 1`] = `ShallowWrapper {}`;
